### PR TITLE
GH-897 Reset Table Data

### PR DESCRIPTION
**Pull request by dtsnell4**
Resetting the table data (.columns([]) and .data([]) ) would cause the widget to error.

Fixes GH-897

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

### CODE REVIEW

**Comment by dtsnell4** [comment]
@GordonSmith @jbrundage @mlzummo Please review

**Comment by dtsnell4** [comment]
http://rawgit.com/dtsnell4/Visualization/GH-897_Table_Reset/demos/dermatology.html?src/other/Table

File: src/other/Table.js [hunk 1]
```diff
@@ -322,9 +322,11 @@
             .remove()
         ;
 
-        if (this.data().length) {
-            this.fixedLabelsUpdate(domNode, element);
+        if (!this.data().length || !this.columns().length) {
+            this.fixedHeader(false);
+            this.fixedColumn(false);
```
Comment (GordonSmith) [inline src/other/Table.js]: If I clear out the data and then load more in - I would expect it to still have fixed headers/columns.

**Comment by dtsnell4** [comment]
@GordonSmith @jbrundage @mlzummo Please re-review.  New strategy.

File: src/other/Table.js [hunk 2]
```diff
@@ -322,10 +322,12 @@
             .remove()
         ;
 
-        if (this.data().length) {
+        d3.select(".rows-wrapper").remove();
+        d3.select(".cols-wrapper").remove();
```
Comment (GordonSmith) [inline src/other/Table.js]: Does this mean the header and column is re-rendered every update cycle?

Comment (dtsnell4) [inline src/other/Table.js]: Good point, let me refactor that.

File: src/other/Table.js [hunk 3]
```diff
@@ -321,11 +321,17 @@
         cells.exit()
             .remove()
         ;
+        if (!this.data().length) {
+            d3.select(".cols-wrapper").remove();        
+        }
+        if (!this.columns().length) {
+            d3.select(".rows-wrapper").remove();
+        }
```
Comment (dtsnell4) [inline src/other/Table.js]: @GordonSmith I made these conditional so they only execute when the appropriate data is resetting.  I may even be over thinking this.  When the data is emptied and then new data is added, the header and footer re-render fine.  These statements are ONLY necessary when the data is emptied and then the table is rendered.  The fixed header and footer code never runs so they are left in the dom.  Is this even a realistic scenario or am I getting carried away?

**Pull request merged**